chainer/dataset/tabular/tabular_dataset.py

```diff
@@ -101,8 +101,8 @@ def slice(self):
         Returns:
             A view of specifed range.
         """
-        from chainer.dataset.tabular.slice import SliceHelper
-        return SliceHelper(self)
+        from chainer.dataset.tabular import slice as slice_
```

Comment:
Not sure but maybe,
`import chainer.dataset.tabular.slice as slice_`
should do the trick.

Also you can move it to the top.

Reply:
I tried it but failed even in py3.

Reply:
Oh that is weird, it works with
`import numpy.random as rand` or similar code.

Reply:
I think the problem is `TabularDataset` requires `Slice/AsTuple/AsDict` where they require `TabularDataset` (cyclic import). Currently, I avoid that by importing them at runtime  (not in initialize time)